src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -61,56 +74,23 @@ public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
     public RuntimeConfig GetConfig()
     {
-        if (_runtimeConfig is not null)
+        if (ConfigLoader.RuntimeConfig is not null)
         {
-            return _runtimeConfig;
+            return ConfigLoader.RuntimeConfig;
         }
 
         // While loading the config file, replace all the environment variables with their values.
-        if (ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? config, replaceEnvVar: true))
-        {
-            _runtimeConfig = config;
-            TrySetupConfigFileWatcher();
-        }
+        bool isLoaded = ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? runtimeConfig, replaceEnvVar: true);
```

Comment:
These last few lines can be condensed and can avoid the `!` null forgiving operator with the following:
```csharp
        if (!ConfigLoader.TryLoadKnownConfig(out RuntimeConfig? runtimeConfig, replaceEnvVar: true))
        {
            throw new DataApiBuilderException(
                message: "Runtime config isn't setup.",
                statusCode: HttpStatusCode.ServiceUnavailable,
                subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
        }

        return runtimeConfig;
```